docs/platform/custom-cloud-credentials/troubleshooting.md

```diff
@@ -0,0 +1,30 @@
+# Bring Your Own Cluster (BYOC) Troubleshooting
+
+## Deploying BYOC
+
+### Kubernetes API Authentication Errors
+AWS STS regional endpoints have to be enabled in the target region. Go to https://console.aws.amazon.com/iam/home#/account_settings and verify the regional endpoint is activated. In most cases your region already has AWS STS regional endpoint enabled -> https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html
+
+!!! note
+    Skipping this step will cause difficult to debug issues. The kubelet will be unable to authenticate against the kubernetes API server, and nothing will work.
+
+## Deploying Workloads 
+
+### Stuck in Pending State
+There are usually three scenarios for why a session or run will be stuck in a pending state: awaiting AWS provisioning, one time start-up process **(sessions only)**, or the instance type is not available in the availability zones your BYOC cluster has access to. We will cover each of the three scenarios in detail here.
+
+#### Sessions One Time Startup Process
+
+When ever a BYOC is deployed there is a one time startup process it must go through before it can deploy sessions. Regardless of the instance type selected this process can take up to 30 minutes. Once completed your BYOC will never have to undergo the startup process again before deploying sessions
```

Comment:
We handle this by not marking cluster as ready until this process completes, see https://linear.app/gridai/issue/GRID-8362/add-start-up-probe-to-the-gridlet-until-the-snapshot-is-not-ready

Reply:
I see that this ticket is in review stage and the associated PR is in draft stage. Once it is merged and released I can update this content to reflect that.

Reply:
Or @nmiculinic is this expected to go into the next release? I want to keep these docs current with the current release

Reply:
@oojo12 how will you keep track to make sure to update this when the linear ticket is merged?

Reply:
Engineering will have to inform me.

Reply:
No, this will not be in next release, it'll be in one after that

Reply:
Just note it'll be released soon, since the fix is already in master